backend/composer/services/export/helpers/export_batch.py

```diff
@@ -81,24 +81,30 @@ def compute_metrics(export_batch: ExportBatch):
     return export_batch
 
 
-def do_transition_to_exported_and_get_available_export_batch(
-    export_batch: ExportBatch, user: User
-):
+def filter_statements_with_exported_transition(qs: QuerySet, user: User) -> QuerySet:
```

Comment:
Conceptually I think this is correct but before approving I would like to profile this with some relatively heavy load.

Reply:
This might be a good use case for [.iterator](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#iterator)